### PR TITLE
Add replace methods to Output builders

### PR DESCRIPTION
**Pull request by thibault-martinez**
Useful for chained output where you just want to change one condition/feature from the previous state.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by bingyanglin — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

File: bee-message/src/output/alias.rs [hunk 1]
```diff
@@ -123,6 +120,19 @@ impl AliasOutputBuilder {
         self
     }
 
+    ///
+    pub fn replace_unlock_condition(mut self, unlock_condition: UnlockCondition) -> Self {
```
Comment (Alex6323) [inline bee-message/src/output/alias.rs]: Would `add_or_replace_unlock_condition` be a better name?

File: bee-message/src/output/alias.rs [hunk 2]
```diff
@@ -137,6 +147,19 @@ impl AliasOutputBuilder {
         self
     }
 
+    ///
+    pub fn replace_feature_block(mut self, feature_block: FeatureBlock) -> Self {
```
Comment (Alex6323) [inline bee-message/src/output/alias.rs]: also here... `add_or_replace_feature_block` is more fitting IMHO.

File: bee-message/src/output/alias.rs [hunk 3]
```diff
@@ -154,6 +177,19 @@ impl AliasOutputBuilder {
         self
     }
 
+    ///
+    pub fn replace_immutable_feature_block(mut self, immutable_feature_block: FeatureBlock) -> Self {
```
Comment (Alex6323) [inline bee-message/src/output/alias.rs]: same here

File: bee-message/src/output/alias.rs (hunk 1 above)
Comment (pvdrz) [reply bee-message/src/output/alias.rs]: what about `upsert_unlock_condition`? too niche?

**Review by pvdrz — DISMISSED** [review]

File: bee-message/src/output/alias.rs [hunk 4]
```diff
@@ -61,8 +54,12 @@ impl AliasOutputBuilder {
         byte_cost_config: ByteCostConfig,
         alias_id: AliasId,
     ) -> Result<AliasOutputBuilder, Error> {
+        Self::new(OutputBuilderAmount::MinimumStorageDeposit(byte_cost_config), alias_id)
+    }
+
+    fn new(amount: OutputBuilderAmount, alias_id: AliasId) -> Result<AliasOutputBuilder, Error> {
```
Comment (Alex6323) [inline bee-message/src/output/alias.rs:60]: Could something like
`fn new(amount: impl Into<OutputBuilderAmount>, alias_id: AliasId)` make the API a bit simpler?

File: bee-message/src/output/alias.rs (hunk 1 above)
Comment (Alex6323) [reply bee-message/src/output/alias.rs]: yea, I like that. Keeps the method name short as before and addresses my suggestion.

Comment (thibault-martinez) [reply bee-message/src/output/alias.rs]: Hmm I think it's really niche tbh, I didn't even know this term a couple of weeks ago...

Comment (thibault-martinez) [reply bee-message/src/output/alias.rs]: I din't have an issue with long method names

File: bee-message/src/output/alias.rs (hunk 4 above)
Comment (thibault-martinez) [reply bee-message/src/output/alias.rs:60]: It's a private method, it doesn't really need to be simpler than that

**Review by grtlr — APPROVED** [review]

**Review by bingyanglin — APPROVED** [review]

**Review by pvdrz — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**